contracts/L2_TokenRetriever.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import { Lockable } from "./Lockable.sol";
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import { WithdrawalAdapter } from "./chain-adapters/l2/WithdrawalAdapter.sol";
+
+interface IBridgeAdapter {
+    function withdrawTokens(WithdrawalAdapter.WithdrawalInformation[] memory) external;
+}
+
+/**
+ * @notice Contract deployed on an arbitrary L2 to act as an intermediate contract for withdrawals from L3 to L1.
+ * @dev Since each network has its own bridging requirements, this contract delegates that logic to a bridge adapter contract
+ * which performs the necessary withdraw action.
+ */
+contract L2_TokenRetriever is Lockable {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    struct TokenPair {
+        address l1Token;
+        address l2Token;
+    }
+
+    // Should be set to the bridge adapter which contains the proper logic to withdraw tokens on
+    // the deployed L2
+    address public immutable bridgeAdapter;
```

Comment:
is "l2WithdrawalAdapter" a better name? makes it clear which chain it exists on and what its function is

Reply:
One of the later comments was about merging the token retriever with the adapters, so this is no longer applicable https://github.com/across-protocol/contracts/pull/605/commits/4db2572d1a4b4bf7ebb81d87659f57e3afda4f0b